hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/viewfs/InodeTree.java

```diff
@@ -413,6 +413,11 @@ public T getTargetFileSystem() throws IOException {
       }
       return targetFileSystem;
     }
+
+    T getTargetFileSystemForClose() throws IOException {
```

Comment:
what difference between this method and `getTargetFileSystem`?

Reply:
Some target FileSystem may not have be initialized. `getTargetFileSystem` initializes all target FileSystems (line1039 below). It's costly and unnecessary when we just try to close them.
https://github.com/apache/hadoop/blob/c44823dadb73a3033f515329f70b2e3126fcb7be/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/viewfs/ViewFileSystem.java#L1036-L1039